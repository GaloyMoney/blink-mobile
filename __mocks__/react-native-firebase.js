/* eslint-disable no-undef */
jest.mock("react-native-firebase", () => {
  return {
    default: {
      messaging: jest.fn(() => ({
        hasPermission: jest.fn(() => Promise.resolve(true)),
        subscribeToTopic: jest.fn(),
        unsubscribeFromTopic: jest.fn(),
        requestPermission: jest.fn(() => Promise.resolve(true)),
        getToken: jest.fn(() => Promise.resolve("myMockToken")),
      })),
      notifications: jest.fn(() => ({
        onNotification: jest.fn(),
        onNotificationDisplayed: jest.fn(),
      })),
      analytics: jest.fn(() => ({
        logEvent: jest.fn(),
      })),
      functions: jest.fn(() => ({
        httpsCallable: jest.fn(),
      })),
    },
  }
})
