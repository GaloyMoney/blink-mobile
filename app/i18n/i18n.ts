import i18n from "i18n-js"

import * as en from "./en.json"
import * as es from "./es.json"

i18n.fallbacks = true
i18n.translations = { en, es }
