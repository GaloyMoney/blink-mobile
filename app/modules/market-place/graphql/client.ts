import {
  ApolloClient,
  ApolloLink,
  DefaultOptions,
  from,
  HttpLink,
  InMemoryCache,
  fromPromise,
} from "@apollo/client"

import { getMainDefinition } from "@apollo/client/utilities"
import { setContext } from "@apollo/client/link/context"
import { onError } from "@apollo/client/link/error"
import { createUploadLink } from "apollo-upload-client"
import { scriptHostname } from "../utils/helper"

const attemps = 0
const GRAPHQL_REGTEST_URI = `http://${scriptHostname()}:4002/graphql`
const GRAPHQL_TESTNET_URI = "https://api.pvbtc1.staging.pvbtc.cloud/graphql"
const GRAPHQL_MAINNET_URI = "https://api.production.pvbtc.cloud/graphql"

const GRAPHQL_REGTEST_WS_URI = `ws://${scriptHostname()}:4002/graphql`
const GRAPHQL_TESTNET_WS_URI = "wss://api.pvbtc1.staging.pvbtc.cloud/graphql"
const GRAPHQL_MAINNET_WS_URI = "wss://api.production.pvbtc.cloud/graphql"
const errorLink = onError(({ graphQLErrors, networkError, operation }) => {
  if (graphQLErrors)
    graphQLErrors.forEach(async ({ message, locations, path }) => {
      switch (message) {
        case "Unauthorized": {
        }
        default:
          console.log(``)
      }
    })
  if (networkError) console.log(`[Network error]: ${networkError}`)
})
const httpLink = createUploadLink({ uri: "https://api-puravida.herokuapp.com/graphql" })
// new HttpLink({
//   uri: `https://api-puravida.herokuapp.com/graphql`,
// })

const authLink = setContext(async (_, { headers }) => {
  // get the authentication token from local storage if it exists
  // const token = localStorage.getItem('token');
  const token = ""
  // return the headers to the context so httpLink can read them
  return {
    headers: {
      ...headers,
      token: token ? `${token}` : "",
    },
  }
})

export const cache = new InMemoryCache({ addTypename: false })

const defaultOptions: DefaultOptions = {
  watchQuery: {
    fetchPolicy: "no-cache",
    errorPolicy: "ignore",
  },
  query: {
    fetchPolicy: "no-cache",
    errorPolicy: "all",
  },
}

const client = new ApolloClient({
  // link: authLink.concat(errorLink.concat(Â©httpLink)),
  link: from([authLink, errorLink, httpLink]),
  cache,
  defaultOptions,
})

export default client
