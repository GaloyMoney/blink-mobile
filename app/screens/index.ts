export * from "./backup-screen"
export * from "./import-wallet-screen"
export * from "./transaction-history"
export * from "./intro-screen"
