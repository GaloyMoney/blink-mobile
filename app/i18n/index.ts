// eslint-disable-next-line import/no-unassigned-import
import "./i18n"

export * from "./translate"
