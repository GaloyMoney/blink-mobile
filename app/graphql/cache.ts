import { InMemoryCache, gql } from "@apollo/client"
import { Account, MyWalletsFragmentDoc, Wallet, WalletCurrency } from "./generated"
import { relayStylePagination } from "@apollo/client/utilities"
import { ReadFieldFunction } from "@apollo/client/cache/core/types/common"

gql`
  fragment MyWallets on ConsumerAccount {
    wallets {
      id
      balance
      walletCurrency
    }
  }

  query realtimePrice {
    me {
      id
      defaultAccount {
        id
        realtimePrice {
          btcSatPrice {
            base
            offset
          }
          denominatorCurrency
          id
          timestamp
          usdCentPrice {
            base
            offset
          }
        }
      }
    }
  }
`

type getWalletsInputs = {
  readField: ReadFieldFunction
  cache: InMemoryCache
}

const getWallets = ({
  readField,
  cache,
}: getWalletsInputs): readonly Wallet[] | undefined => {
  const id = readField("id")
  const key = `ConsumerAccount:${id}`
  const account: Account | null = cache.readFragment({
    id: key,
    fragment: MyWalletsFragmentDoc,
  })
  if (account === null) {
    return undefined
  }
  return account.wallets
}

export const createCache = () =>
  new InMemoryCache({
    possibleTypes: {
      // TODO: add other possible types
      Wallet: ["BTCWallet", "UsdWallet"],
      Account: ["ConsumerAccount"],
    },
    typePolicies: {
      Globals: {
        // singleton: only cache latest version:
        // https://www.apollographql.com/docs/react/caching/cache-configuration/#customizing-cache-ids
        keyFields: [],
      },
      RealtimePrice: {
        keyFields: [],
      },
      MapMarker: {
        keyFields: ["mapInfo", ["title", "coordinates"]],
      },
      Contact: {
        fields: {
          prettyName: {
            read(_, { readField }) {
              return readField("id") || readField("name")
            },
          },
        },
      },
      UserContact: {
        fields: {
          transactions: relayStylePagination(),
        },
      },
      Earn: {
        fields: {
          completed: {
            read: (value) => value ?? false,
          },
        },
      },
      Query: {
        fields: {
          // local only fields
          hideBalance: {
            read: (value) => value ?? false,
          },
          hiddenBalanceToolTip: {
            read: (value) => value ?? false,
          },
          beta: {
            read: (value) => value ?? false,
          },
          colorScheme: {
            read: (value) => value ?? "light",
          },
        },
      },
      ConsumerAccount: {
        fields: {
          usdWallet: {
            read: (_, { readField, cache }): Wallet | undefined => {
              const wallets = getWallets({ readField, cache })
              if (wallets === undefined || wallets.length === 0) {
                return undefined
              }

              // TODO: return toReference instead
              // https://www.apollographql.com/docs/react/caching/advanced-topics#cache-redirects
              return wallets.find(
                (wallet) => wallet.walletCurrency === WalletCurrency.Usd,
              )
            },
          },
          btcWallet: {
            read: (_, { readField, cache }) => {
              const wallets = getWallets({ readField, cache })
              if (wallets === undefined || wallets.length === 0) {
                return undefined
              }

              // TODO: return toReference instead
              // https://www.apollographql.com/docs/react/caching/advanced-topics#cache-redirects
              return wallets.find(
                (wallet) => wallet.walletCurrency === WalletCurrency.Btc,
              )
            },
          },
          defaultWallet: {
            read: (_, { readField, cache }): Wallet | undefined => {
              const wallets = getWallets({ readField, cache })
              if (wallets === undefined || wallets.length === 0) {
                return undefined
              }

              const defaultWalletId = readField("defaultWalletId")

              // TODO: return toReference instead
              // https://www.apollographql.com/docs/react/caching/advanced-topics#cache-redirects
              return wallets.find((wallet) => wallet.id === defaultWalletId)
            },
          },
          transactions: relayStylePagination(),
        },
      },
      Wallet: {
        fields: {
          transactions: relayStylePagination(),
        },
      },
    },
  })
