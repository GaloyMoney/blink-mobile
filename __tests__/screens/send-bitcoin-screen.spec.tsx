import * as React from "react"
import { MockedProvider } from "@apollo/client/testing"
import { InMemoryCache } from "@apollo/client"
import { act, cleanup, fireEvent, render } from "@testing-library/react-native"
import renderer from "react-test-renderer"
import "@testing-library/jest-native/extend-expect"
import "@node_modules/react-native-gesture-handler/jestSetup.js"

import { translate } from "@app/i18n/translate"
import "@mocks/react-native-firebase"
import "@mocks/react-navigation-native"
import { SendBitcoinScreen } from "@app/screens/send-bitcoin-screen"
import { waitForNextRender } from "../helpers/wait"
import { cacheNodeStats, cachePrice, cacheUsername, cacheWallet } from "../helpers/cache"

jest.mock("../../app/utils/parsing", () => {
  const actualParsing = jest.requireActual("../../app/utils/parsing")
  return {
    ...actualParsing,
    lightningInvoiceHasExpired: () => false,
  }
})

afterEach(cleanup)

describe("SendBitcoinScreen", () => {
  const cache = new InMemoryCache()
  cacheWallet(cache, 117585)
  cacheNodeStats(cache)
  cachePrice(cache)
  cacheUsername(cache)

  it("render matches snapshot", () => {
    const tree = renderer
      .create(
        <MockedProvider cache={cache}>
          <SendBitcoinScreen route={{ params: null }} />
        </MockedProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has TextInputs", () => {
    const { queryByA11yLabel, queryByPlaceholderText } = render(
      <MockedProvider cache={cache}>
        <SendBitcoinScreen route={{ params: null }} />
      </MockedProvider>,
    )

    expect(queryByA11yLabel("Input payment")).not.toBeNull()
    expect(queryByPlaceholderText(translate("SendBitcoinScreen.input"))).not.toBeNull()
    expect(queryByPlaceholderText(translate("SendBitcoinScreen.note"))).not.toBeNull()
  })

  it("shows send only when an amount and destination are present", () => {
    const { getByA11yLabel, getByPlaceholderText, queryByText } = render(
      <MockedProvider cache={cache}>
        <SendBitcoinScreen route={{ params: null }} />
      </MockedProvider>,
    )

    const amountInput = getByA11yLabel("Input payment")
    const destinationInput = getByPlaceholderText(translate("SendBitcoinScreen.input"))

    expect(queryByText(translate("common.amountRequired"))).not.toBeNull()
    expect(queryByText(translate("common.usernameRequired"))).toBeNull()
    expect(queryByText(translate("common.send"))).toBeNull()

    fireEvent(amountInput, "onChangeText", "27226")

    expect(queryByText(translate("common.amountRequired"))).toBeNull()
    expect(queryByText(translate("common.usernameRequired"))).not.toBeNull()
    expect(queryByText(translate("common.send"))).toBeNull()

    fireEvent.changeText(destinationInput, "Bitcoin")

    expect(queryByText(translate("common.amountRequired"))).toBeNull()
    expect(queryByText(translate("common.usernameRequired"))).toBeNull()
    expect(queryByText(translate("common.send"))).not.toBeNull()
  })

  it("shows error when total exceeds the balance", () => {
    const { getByA11yLabel, queryByText } = render(
      <MockedProvider cache={cache}>
        <SendBitcoinScreen route={{ params: null }} />
      </MockedProvider>,
    )

    const amountInput = getByA11yLabel("Input payment")

    expect(
      queryByText(translate("SendBitcoinScreen.amountExceed", { balance: "$46.64" })),
    ).toBeNull()

    fireEvent(amountInput, "onChangeText", "27226")

    expect(
      queryByText(translate("SendBitcoinScreen.amountExceed", { balance: "$46.64" })),
    ).not.toBeNull()

    fireEvent(amountInput, "onChangeText", "2722")

    expect(
      queryByText(translate("SendBitcoinScreen.amountExceed", { balance: "$46.64" })),
    ).toBeNull()
  })

  it("successfully parses lightning payment with `lightning:` prefix", async () => {
    cacheWallet(cache, 1175855)

    const { getByA11yLabel, getByPlaceholderText, queryByText } = render(
      <MockedProvider cache={cache}>
        <SendBitcoinScreen route={{ params: null }} />
      </MockedProvider>,
    )

    const displayText = getByA11yLabel("Input payment display value")
    const destinationInput = getByPlaceholderText(translate("SendBitcoinScreen.input"))

    fireEvent.changeText(
      destinationInput,
      "lightning:lnbc6864270n1p05zvjjpp5fpehvlv3dd2r76065r9v0l3n8qv9mfwu9ryhvpj5xsz3p4hy734qdzhxysv89eqyvmzqsnfw3pxcmmrddpx7mmdypp8yatwvd5zqmmwypqh2em4wd6zqvesyq5yyun4de3ksgz0dek8j2gcqzpgxqrrss6lqa5jllvuglw5tpsug4s2tmt5c8fnerr95fuh8htcsyx52cp3wzswj32xj5gewyfn7mg293v6jla9cz8zndhwdhcnnkul2qkf6pjlspj2nl3j",
    )

    await act(await waitForNextRender)
    expect(displayText).toHaveTextContent("272.26")

    expect(queryByText(translate("common.send"))).not.toBeNull()
  })
})
