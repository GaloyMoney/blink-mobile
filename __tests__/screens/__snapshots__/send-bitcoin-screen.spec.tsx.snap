// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendBitcoinScreen render matches snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#E6EBEf",
          "flex": 1,
          "height": "100%",
        },
        Object {},
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    onSwipeComplete={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection={
      Array [
        "down",
      ]
    }
    swipeThreshold={50}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipeComplete={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "flexGrow": 1,
          "justifyContent": "center",
          "left": 0,
          "margin": 37.5,
          "marginBottom": 0,
          "marginHorizontal": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeDirection={
        Array [
          "down",
        ]
      }
      swipeThreshold={50}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "space-around",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        />
      </View>
      <RNCSafeAreaView
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "paddingHorizontal": 20,
            "paddingTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "height": 14,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#CFD9E2",
                  "fontSize": 72,
                },
                Object {
                  "height": 40,
                  "top": -40,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "fontSize": 18,
              "marginVertical": 8,
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "alignContent": "stretch",
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#3050C4",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginHorizontal": 20,
                    "marginVertical": 10,
                    "padding": 8,
                    "width": 145,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Dismiss
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderRadius": 3,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 3,
                },
                undefined,
                false,
              ]
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#3050C4",
                    "borderColor": "#2089dc",
                    "borderRadius": 3,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginHorizontal": 20,
                    "marginVertical": 10,
                    "padding": 8,
                    "width": 145,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Open
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RNCSafeAreaView>
    </View>
  </Modal>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        Object {},
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "justifyContent": "space-between",
        }
      }
    >
      <View>
        <View>
          <View>
            <View>
              <Text
                ellipsizeMode="middle"
                numberOfLines={1}
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                0.00
              </Text>
              <View
                style={
                  Object {
                    "paddingHorizontal": 10,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#86939e",
                      "flexDirection": "row",
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "height": 40,
                        "justifyContent": "center",
                        "marginVertical": 4,
                        "paddingRight": 4,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          undefined,
                          Object {
                            "color": "#939AA4",
                          },
                        ]
                      }
                    >
                      $
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoFocus={true}
                    contextMenuHidden={true}
                    editable={true}
                    enablesReturnKeyAutomatically={true}
                    keyboardType="number-pad"
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onEndEditing={[Function]}
                    placeholderTextColor="#86939e"
                    rejectResponderTermination={true}
                    returnKeyLabel="Update"
                    returnKeyType="done"
                    selection={
                      Object {
                        "end": 4,
                        "start": 4,
                      }
                    }
                    style={
                      Object {
                        "color": "#242424",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                    value="0.00"
                  />
                </View>
                <Text
                  style={
                    Object {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
              <RNGestureHandlerButton
                collapsable={false}
                onGestureEvent={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onHandlerStateChange={[Function]}
                rippleColor={0}
              >
                <View
                  accessible={true}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                          "fontSize": 32,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </RNGestureHandlerButton>
            </View>
            <Text>
              0
               
              sats
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 18,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View>
                  <Text>
                    To
                  </Text>
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#3050C4",
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                autoCompleteType="username"
                editable={false}
                onChangeText={[Function]}
                placeholder="username or invoice"
                placeholderTextColor="#86939e"
                rejectResponderTermination={true}
                selectTextOnFocus={true}
                style={
                  Object {
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 30,
                      },
                      undefined,
                      Object {
                        "fontFamily": "Ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View>
                  <Text>
                    Note
                  </Text>
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#3050C4",
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                onChangeText={[Function]}
                placeholder="optional note"
                placeholderTextColor="#86939e"
                rejectResponderTermination={true}
                selectTextOnFocus={true}
                style={
                  Object {
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 10,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#86939e",
                  "flexDirection": "row",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View>
                  <Text>
                    Fee
                  </Text>
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#3050C4",
                          "fontSize": 24,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Ionicons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={false}
                placeholder="network fee"
                placeholderTextColor="#86939e"
                rejectResponderTermination={true}
                selectTextOnFocus={true}
                style={
                  Object {
                    "color": "#242424",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 40,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#ff190c",
                  "fontSize": 16,
                  "height": 18,
                  "margin": 5,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 3,
              },
              Object {
                "flex": 1,
              },
              false,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "hsl(208, 8%, 90%)",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "hsl(208, 8%, 63%)",
                    "fontSize": 18,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Amount is required
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
